calculators/referendum_expenditures_by_origin.rb

```diff
@@ -0,0 +1,101 @@
+class ReferendumExpendituresByOrigin
+  def initialize(candidates: [], ballot_measures: [], committees: [])
+    @ballot_measures = ballot_measures
+    @committees_by_filer_id =
+      committees.where('"Filer_ID" IS NOT NULL').index_by { |c| c.Filer_ID }
+  end
+
+  def fetch
+    totals = ActiveRecord::Base.connection.execute(<<-SQL)
+      SELECT "Measure_Number", "Sup_Opp_Cd", sum("Amount") AS total
+      FROM "efile_COAK_2016_E-Expenditure",
+        oakland_name_to_number
+      WHERE "Bal_Name" = "Measure_Name"
+      GROUP BY "Measure_Number", "Sup_Opp_Cd";
```

Comment:
Im not so sure your technique works right.  We can have multiple 460 filings each with their own sequences of Report_Num's.  if we group on Report_Num then each of those groups are only for the 460s that have that Report_Num.  e.g. say there are filings in Jan and June.  The Jan Filing has 1 amendment and the June has 2.  The aggregation will be segmented such that the Report_Num = 2 group will only have the June filing and will not include any data from Jan.